V2rayNG/app/src/main/kotlin/com/v2ray/ang/util/V2rayConfigUtil.kt

```diff
@@ -161,11 +161,8 @@ object V2rayConfigUtil {
 //                v2rayConfig.inbounds.add(httpCopy)
 //            }
 
-            if (!app.defaultDPreference.getPrefBoolean(SettingsActivity.PREF_LOCAL_DNS_ENABLED, false)) {
-                v2rayConfig.inbounds[0].sniffing?.enabled = true
-            } else {
-                v2rayConfig.inbounds[0].sniffing?.enabled = app.defaultDPreference.getPrefBoolean(SettingsActivity.PREF_SNIFFING_ENABLED, true)
-            }
+            // enforce trafficSniffing to prevent from dns-poisoning
+            v2rayConfig.inbounds[0].sniffing?.enabled = true
```

Comment:
this doesn't make sense. Sending dns traffic to server also mitigate dns-poisoning. I think someone may not want sniffing function.

Reply:
sniffing had to enabled, or you will met dns-poisoning, you can try it.

Reply:
https://github.com/2dust/v2rayNG/issues/653